media-source/MediaSource-duration.html

```diff
@@ -0,0 +1,84 @@
+<!doctype html>
+<html>
+<head>
+  <meta charset='utf-8'>
+  <title>MediaSource#duration</title>
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+</head>
+<body>
+<div id="log"></div>
+<script>
+var video = document.createElement('video');
+var contents = {'/media/white.webm': 'video/webm; codecs="vorbis,vp8"',
+                '/media/white.mp4' : 'video/mp4'};
+
+//check the browser supports the MIME used in this test
+function isTypeSupported(mime) {
+    if(!MediaSource.isTypeSupported(mime)) {
+        this.step(function() {
+            assert_unreached("Browser doesn't support the MIME used in this test: " + mime);
+        });
+        this.done();
+        return false;
+    }
+    return true;
+}
+function GET(url, processBody) {
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', url, true);
+    xhr.responseType = 'arraybuffer';
+    xhr.send();
+    xhr.onload = function(e) {
+        if (xhr.status != 200) {
+            alert("Unexpected status code " + xhr.status + " for " + url);
+            return false;
+        }
+        processBody(new Uint8Array(xhr.response));
+    };
+}
+test(function() {
+    var mediaSource = new MediaSource();
+    assert_equals(mediaSource.duration, NaN);
+    assert_throws({name : 'InvalidAccessError'},
+                  function() {
+                      mediaSource.duration = -1;
+                  }, 'Set invalid duration');
+    assert_throws({name : 'InvalidAccessError'},
+                  function() {
+                      mediaSource.duration = NaN;
+                  }, 'Set invalid duration');
```

Comment:
Invalid duration settting is covered already by mediasource-duration-boundaryconditions.html, though not necessarily when in closed state. I think there's sufficient overlap that this particular set of tests in this file are not needed.